docs/android/codestyle.md

```diff
@@ -0,0 +1,70 @@
+---
+title: "Android Code Style"
+sidebar_label: "Code Style"
+---
+
+## Why Enforcing Code Style
+
+We aim to maintain a **consistent and standard codebase**. By enforcing code style:
+
+- We reduce unnecessary comments on PRs, allowing reviewers to focus on the logic rather than formatting.
+- We ensure the code is easier to read and maintain.
+
+## Language Guidelines
+
+- All code must be written in **English**.
+- Avoid typos and grammatical mistakes. While mistakes are acceptable (as many contributors are non-native speakers), reviewers are encouraged to suggest corrections.
+- Use tools like LLMs or spell checkers to assist in correcting mistakes.
+
+## KTLint Formatter
+
+We use [KTLint](https://pinterest.github.io/ktlint) to enforce Kotlin code style. It is integrated into our Gradle modules and configured via the `.editorconfig` file.
+
+### Custom Rules
+
+- We override some KTLint rules when necessary. To enable or disable a rule:
+  1. Submit a **dedicated PR** with a proper explanation of the change.
+  2. If the change affects a large portion of the codebase, create **two commits**:
+     - One for updating the rule.
+     - Another for applying the changes.
+
+### Running KTLint
+
+You can use KTLint through Gradle to automatically reformat your code:
+
+```bash
+./gradlew :build-logic:convention:ktlintFormat ktlintFormat
+```
+
+### CI Integration
+
+If a KTLint error is detected, the CI will fail, and GitHub will report it as a comment in the PR using the generated [SARIF](tips/sarif_reports.md) report.
+
+## Yamllint
+
+We use [Yamllint](https://github.com/adrienverge/yamllint) to enforce YAML formatting. The `github` format is followed for all YAML files in the repository.
+
+### Running Yamllint
+
+Run the following command at the root of the repository to check YAML formatting:
+
+```bash
+yamllint --strict --format github .
+```
+
+- **Note:** Yamllint does not reformat your code; it only reports errors that need to be fixed.
+- Use your IDE's code formatter or fix issues manually.
+
+### CI Integration
+
+If the YAML format is invalid, the CI will block the PR.
+
+## TODOs Are Tracked in GitHub
+
+When working on a PR, you may want to add a `TODO` for later. To properly track TODOs, associate them with a GitHub issue.
```

Comment:
I feel like this encourages people to create issues before submitting a PR, and incomplete code. The end result will be issues we have no context for, which potentially will never be used, and a deviation from the normal review process.

Reply:
Yeah I was afraid of this. Do you have an idea of how we could write this. Like we don't want TODO in the code but if we do add one it needs to be agreed on the PR and tracked with an issue?

Reply:
Add a review comment if you feel like it needs to be expanded/changed later? Then it can be discussed, and if needed an issue can be created once merged. But it's also very much possible that the outcome is that it should be handled before the PR is merged, or the code is changed so much that it is no longer relevant.

Reply:
Updated it with a better phrasing.